chemprop/hyperopt_utils.py

```diff
@@ -18,11 +50,32 @@ def merge_trials(trials: Trials, new_trials_data: List[Dict]) -> Trials:
     :param trials_data: The contents of a hyperopt trials object, `Trials.trials`.
     :return: A hyperopt trials object, merged from the two inputs.
     """
-    max_tid = 0
     if len(trials.trials) > 0:
         max_tid = max([trial['tid'] for trial in trials.trials])
+        trial_keys = list(trials.vals.keys())
+        for trial in trials.trials:
+            new_trial_keys = list(trial['misc']['vals'].keys())
+            if trial_keys != new_trial_keys:
```

Comment:
```suggestion
            if set(trial_keys) != set(new_trial_keys):
```
The order of keys in `new_trial_keys` and `trial_keys` would be different. The order is not guaranteed in Python 3.5 and earlier but is the same as the insertion order in Python 3.6 and older. It is generally a better way to handle this with set(). Similar problem is in L73.

Reply:
Right you are. I made these lists into sets instead, not at the point of comparison but at the point of creation.